### PR TITLE
Support xclip, fix #9

### LINKED ISSUE #9: [feature] support more clipper: xsel, ...
See <https://github.com/neovim/neovim/blob/db407010facc55c19b5ebdf881225ac39cb29d01/runtime/autoload/provider/clipboard.vim#L69-L158>

xclip is not the only clipper in linux. It should be not a depend but an optdepend. TIA!

```sh
❯ pacman -Qi perl-clipboard
Name            : perl-clipboard
Version         : 0.28-1
Description     : Perl clipboard module
Architecture    : any
URL             : https://metacpan.org/release/Clipboard
Licenses        : PerlArtistic  GPL
Groups          : None
Provides        : None
Depends On      : perl-cgi  perl-uri  xclip
Optional Deps   : None
Required By     : perl-app-nopaste
Optional For    : None
```

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

**Comment by rwp0** [comment]
I think the title could be better desribed as _Support xsel_ rather than xclip since the latter is already supported by the distribution.

File: lib/Clipboard/Xsel.pm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    open my $exe, $cmd or die "Couldn't run `$cmd`: $!\n";
+    my $result = join '', <$exe>;
+    close $exe or die "Error closing `$cmd`: $!";
+    return $result;
+}
+# This ordering isn't officially verified, but so far seems to work the best:
+sub all_selections { qw(primary buffer clipboard secondary) }
+sub favorite_selection { my $self = shift; ($self->all_selections)[0] }
+
+sub xclip_available {
```
Comment (rwp0) [inline lib/Clipboard/Xsel.pm]: Shouldn't this function be named `xsel_available` instead

File: lib/Clipboard.pm [hunk 2]
```diff
@@ -34,8 +37,9 @@ sub find_driver {
         # available, use it.
         if (exists $ENV{SSH_CONNECTION}) {
             local $SIG{__WARN__} = sub {};
+            require Clipboard::Xsel;
+            return 'Xsel' if Clipboard::Xsel::xsel_available();
```
Comment (rwp0) [inline lib/Clipboard.pm:41]: The reference doesn't match the definition (`xclip_available`) here

File: lib/Clipboard/Xsel.pm [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    open STDERR, '>&', $olderr;
+    close $olderr;
+
+    return $open_retval;
+}
+
+{
+  xclip_available() or warn <<'EPIGRAPH';
+
+Can't find the 'xsel' script.  Clipboard.pm's X support depends on it.
```
Comment (rwp0) [inline lib/Clipboard/Xsel.pm]: xsel isn't a script but a binary executable

**Review by rwp0 — COMMENTED** [review]
Might add tests here for the xsel driver https://github.com/shlomif/Clipboard/blob/master/t/drivers.t

File: lib/Clipboard/Xsel.pm (hunk 1 above)
Comment (Freed-Wu) [reply lib/Clipboard/Xsel.pm]: Aha, it is my fault.

**Comment by Freed-Wu** [comment]
> Might add tests here

I am not familiar with unit test of perl :cry: Can you help?

**Comment by rwp0** [comment]
> > Might add tests here
> 
> I am not familiar with unit test of perl :cry: Can you help?

You might start by reading the `Test::Simple` and then `Test::More` module documentation in Perldoc

**Comment by shlomif** [comment]
@Freed-Wu : seems mostly fine.

**Pull request closed without merging**